### PR TITLE
release v3.3

### LINKED ISSUE #4017: docs: upgrade guide from v3.2 to v3.3
Steps:
1- Cookiecut an instance. 
2- Edit the Pipfile to use the Invenio v3.2.2 module. Install from git or local checkout of https://github.com/inveniosoftware/invenio/pull/4015
3- Follow the quickstart (Okay you already cookiecutted, but you have to bootstrap, setup and server)
4- Create some records and so on. Follow: https://invenio.readthedocs.io/en/latest/quickstart/crud-operations.html
5- Try to upgrade the instance to v3.3. Basically you should change the invenio version in the Pipfile. In this case use: https://github.com/inveniosoftware/invenio/pull/4013. You might have to do extra steps (e.g. remove lxml restriction, etc.)

The task consists on documenting every singly change that needs to be made to update from v3.2.2 to v.3.3

Similar to https://github.com/inveniosoftware/invenio/blob/edb863d2f5228fb158c090a69c2db7c3385b6ba3/docs/upgrading.rst

### PR DESCRIPTION
**Pull request by ppanero**
Migrated from https://github.com/inveniosoftware/invenio/issues/4002

Closes #4017
Closes https://github.com/inveniosoftware/invenio/issues/4008
Closes https://github.com/inveniosoftware/invenio/issues/4011
Closes https://github.com/inveniosoftware/invenio/issues/4010

### CODE REVIEW

**Comment by ppanero** [comment]
Closes https://github.com/inveniosoftware/invenio-oaiserver/issues/184

File: setup.py [hunk 1]
```diff
@@ -105,14 +102,12 @@
...
 install_requires = [
-    'Flask>=1.0.4',
-    'invenio-app>=1.2.3,<1.3.0',
-    'invenio-base>=1.2.0,<1.3.0',
-    'invenio-cache>=1.0.0,<1.1.0',
-    'invenio-celery>=1.1.1,<1.2.0',
-    'invenio-config>=1.0.2,<1.1.0',
-    'invenio-i18n>=1.1.1,<1.2.0',
-    'Werkzeug>=0.15.0,<1.0.0'
+    'invenio-app>=1.2.6,<1.3.0',
```
Comment (ppanero) [inline setup.py:105]: comment: it is okay. **must not be bumped**, invenio-app 1.3.0+ contains semantic-ui work.

File: setup.py [hunk 2]
```diff
@@ -20,46 +20,46 @@
...
         'invenio-theme>=1.1.4,<1.2.0',
     ],
     'auth': [
-        'invenio-access>=1.3.0,<1.4.0',
-        'invenio-accounts>=1.1.1,<1.2.0',
-        'invenio-oauth2server>=1.0.4,<1.1.0',
-        'invenio-oauthclient>=1.1.3,<1.2.0',
-        'invenio-userprofiles>=1.0.1,<1.1.0',
+        'invenio-access>=1.4.1,<1.5.0',
+        'invenio-accounts>=1.2.1,< 1.3.0',
```
Comment (ppanero) [inline setup.py:44]: bump. 1.2.2 is released.

File: setup.py [hunk 3]
```diff
@@ -20,46 +20,46 @@
...
-search_version = '>=1.2.3,<1.3.0'
+db_version = '>=1.0.5,<1.1.0'
+search_version = '>=1.3.1,<1.4.0'
 
 extras_require = {
     # Bundles
     'base': [
-        'invenio-admin>=1.1.2,<1.2.0',
+        'invenio-admin>=1.2.0,<1.3.0',
         'invenio-assets>=1.1.3,<1.2.0',
```
Comment (ppanero) [inline setup.py:35]: comment: it is okay, keep >=1.1.3,<1.2.0. must not be bumped, invenio-assets 1.2.0+ contains semantic-ui work.

**Comment by ppanero** [comment]
**MOVE PR TO MERGE AGAINS maint-3.3**

File: docs/releases/v3.3.0.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Invenio v3.3 supports Python version 3.6. Python 2 support in Invenio
+ended on January 1st, 2020 with the official end of life for Python 2 on the
+same date.
+
+Getting started
+---------------
+
+See our :ref:`quickstart` guide.
+
+What's new in Invenio v3.3?
```
Comment (ppanero) [inline docs/releases/v3.3.0.rst:26]: Python 3.7 support

File: setup.py [hunk 5]
```diff
@@ -145,11 +145,10 @@
         'Programming Language :: Python',
         'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
         'Topic :: Software Development :: Libraries :: Python Modules',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
```
Comment (ppanero) [inline setup.py:149]: remove v3.5?

**Comment by lnielsen** [comment]
We have to remember to update this one on Monday: https://invenio.readthedocs.io/en/latest/general/requirements.html

**Pull request merged**